apps/aechannel/src/aesc_state_cache.erl

```diff
@@ -0,0 +1,89 @@
+-module(aesc_state_cache).
+-behaviour(gen_server).
+
+-export([
+          start_link/0
+        , new/3
+        , update/3
+        , fetch/2
+        , delete/1
+        ]).
+
+-export([
+          init/1
+        , handle_call/3
+        , handle_cast/2
+        , handle_info/2
+        , terminate/2
+        , code_change/3
+        ]).
+
+-record(st, {}).
+-record(ch, {id, state}).
+
+-define(SERVER, ?MODULE).
+-define(TAB, ?MODULE).
+
+new(ChId, PubKey, State) ->
+    gen_server:call(?SERVER, {new, ChId, PubKey, State}).
+
+update(ChId, PubKey, State) ->
+    gen_server:cast(?SERVER, {update, ChId, PubKey, State}).
+
+fetch(ChId, PubKey) ->
+    gen_server:call(?SERVER, {fetch, ChId, PubKey}).
+
+delete(ChId) ->
+    gen_server:cast(?SERVER, {delete, ChId}).
+
+start_link() ->
+    case ets:info(?TAB, name) of
+        undefined ->
+            ets:new(?MODULE, [ordered_set, public, named_table,
+                              {keypos, #ch.id}]);
+        _ ->
+            ok
+    end,
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+init([]) ->
+    {ok, #st{}}.
+
+handle_call({new, ChId, PubKey, State}, _From, St) ->
+    case ets:insert_new(?TAB, #ch{id = key(ChId, PubKey), state = State}) of
```

Comment:
I see memory based. Maybe worth a metric so operator has feeling of memory used by this?